source/common/network/lc_trie.h

```diff
@@ -99,11 +99,23 @@ class LcTrie {
   typedef uint32_t Ipv4;
   typedef absl::uint128 Ipv6;
 
+  typedef std::unordered_set<std::string> TagSet;
+  typedef std::shared_ptr<std::unordered_set<std::string>> TagSetSharedPtr;
+
   /**
    * Structure to hold a CIDR range and the tag associated with it.
    */
   template <class IpType, uint32_t address_size = CHAR_BIT * sizeof(IpType)> struct IpPrefix {
 
+    IpPrefix() {}
+
+    IpPrefix(const IpType& ip, int length, const std::string& tag) : ip_(ip), length_(length) {
```

Comment:
Nit: prefer s/int/uint32_t/ here and elsewhere as per Envoy coding idiom (use fixed size types where we're not conflicting with other type conventions, e.g. syscall). I realize this is an extant issue in the member field type, but this is my preference.